common/lib/transport/connectionmanager.js

```diff
@@ -90,6 +79,17 @@ var ConnectionManager = (function() {
 		EventEmitter.call(this);
 		this.realtime = realtime;
 		this.options = options;
+		states = {
```

Comment:
What's the purpose of reinitialising this each time a `ConnectionManager` is initialised, when it's reinitialising something with global scope? Either the structure is immutable, in which case it can be initialised once, or it's mutable (with per-instance state), in which case it should have per-instance scope.